python/hsfs/core/rondb_engine.py

```diff
@@ -0,0 +1,238 @@
+#
+#   Copyright 2024 Hopsworks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+from typing import Optional, Any, Union
+from datetime import datetime
+
+from hsfs.core import rondb_rest_api
+from hsfs import util
+
+
+class RondbEngine:
+    RETURN_TYPE_FEATURE_VECTOR = "feature_vector"
+    RETURN_TYPE_RESPONSE_JSON = "response_json"  # as a python dict
+
+    def __init__(self):
+        self._rondb_rest_api = rondb_rest_api.RondbRestApi()
+
+    def _build_base_payload(
+        self,
+        feature_store_name: str,
+        feature_view_name: str,
+        feature_view_version: int,
+        metadata_options: Optional[dict[str, bool]] = None,
+        return_type: str = RETURN_TYPE_FEATURE_VECTOR,
+    ) -> dict[str, Union[str, dict[str, bool]]]:
+        """Build the base payload for the RonDB REST Server Feature Store API.
+
+        Check the RonDB Rest Server Feature Store API documentation for more details:
+        https://docs.hopsworks.ai/latest/user_guides/fs/feature_view/feature-server
+
+        !!! warning
+            featureName and featureType must be set to True to allow the response to be converted
+            to a feature vector with convert_rdrs_response_to_dict_feature_vector.
+
+        Args:
+            feature_store_name: The name of the feature store in which the feature view is registered.
+                The suffix '_featurestore' should be omitted.
+            feature_view_name: The name of the feature view from which to retrieve the feature vector.
+            feature_view_version: The version of the feature view from which to retrieve the feature vector.
+            metadata_options: Whether to include feature metadata in the response.
+                Keys are "featureName" and "featureType" and values are boolean.
+            return_type: The type of the return value. Either "feature_vector" or "response_json".
+                If "feature_vector" is selected the payload will enforce fetching feature metadata.
+
+        Returns:
+            The payload to send to the RonDB REST Server Feature Store API.
+        """
+        return {
+            "featureStoreName": util.strip_feature_store_suffix(feature_store_name),
+            "featureViewName": feature_view_name,
+            "featureViewVersion": feature_view_version,
+            "metadataOptions": {
+                "featureName": True
```

Comment:
I will add it to the list of possible improvement once the main PR is merged. I am not sure of the gain in terms of performance/efficiency since json anyways has the field and they are simply set to null. Worth maybe asking the rondb people if it costs much in the backend though